### PR TITLE
[KOGITO-9078] [KSW-Operator] Create the cekit image descriptor for Op…

### PR DESCRIPTION
**Pull request by davidesalerno**
…erator image

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [x] Add or Modify a unit test for your change
- [x] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: utils/resources/resources.go [hunk 1]
```diff
@@ -27,11 +27,14 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/kiegroup/kogito-serverless-operator/container-builder/client"
+	"github.com/kiegroup/kogito-serverless-operator/utils"
 
 	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
 	"github.com/kiegroup/kogito-serverless-operator/utils/kubernetes"
 )
 
+var builderResourcesPath = utils.GetEnv("BUILDER_RESOURCES_PATH", "/usr/local/etc/serverless-operator")
```
Comment (davidesalerno) [inline utils/resources/resources.go:36]: This will be usefull especially in local testing

File: Dockerfile [hunk 2]
```diff
@@ -25,14 +25,12 @@ COPY version/ version/
...
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
-#FROM registry.access.redhat.com/ubi8/ubi-micro:latest
-WORKDIR /
-COPY --from=builder /workspace/manager .
-COPY --from=builder /workspace/resources/ ./resources/
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
```
Comment (spolti) [inline Dockerfile:28]: we probably can evaluate the usage of the ubi8-micro image, do you mind testing it?

File: image.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  from: golang:1.19
+  description: Builder Image for the Operator
+
+  modules:
+    repositories:
+      - path: modules
+    install:
+      - name: org.kie.kogito.goModDownload
+      - name: org.kie.kogito.app.builder
+- name: kogito-serverless-operator
```
Comment (spolti) [inline image.yaml:12]:
```suggestion


- name: kogito-serverless-operator
```

Comment (spolti) [reply image.yaml:12]: not mandatory, just to make easier to find where one image definition ends and the other starts.

File: utils/resources/resources.go (hunk 1 above)
Comment (spolti) [reply utils/resources/resources.go:36]: are there any use cases?

Comment (spolti) [reply utils/resources/resources.go:36]: we need to document it as well, I guess

**Review by spolti — APPROVED** [review]
Nice work @davidesalerno 
added a few comments, not blockers :)

Thanks

File: Dockerfile (hunk 2 above)
Comment (davidesalerno) [reply Dockerfile:28]: ok

File: utils/resources/resources.go (hunk 1 above)
Comment (davidesalerno) [reply utils/resources/resources.go:36]: Yes, if you need to execute the Operator locally in debug, you can use that ENV variable to a different folder on your machine (for example your project workspace) where those files are present.

**Review by desmax74 — APPROVED** [review]

File: osl-image-prod.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+- schema_version: 1
+  name: "operator-builder"
+  version: "1.29.0"
+  from: "registry-proxy.engineering.redhat.com/rh-osbs/rhel8-go-toolset:1.19.4-7"
```
Comment (rgdoliveira) [inline osl-image-prod.yaml:4]: I've just checked and the official release of go-toolset is out. There are both 1.19.6 and 1.19.9 versions available.

So you can use either
`from: "registry.access.redhat.com/ubi8/go-toolset:1.19.6"`
or
`from: "registry.access.redhat.com/ubi8/go-toolset:1.19.9"`

File: utils/resources/resources.go (hunk 1 above)
Comment (spolti) [reply utils/resources/resources.go:36]: sounds good, can you please add this info in the README?

File: osl-image-prod.yaml (hunk 4 above)
Comment (spolti) [reply osl-image-prod.yaml:4]: +1 from me on using the latest.

File: Dockerfile (hunk 2 above)
Comment (davidesalerno) [reply Dockerfile]: The ubi8-micro is working well both for dev mode that for production mode, I think we can use that.
I did the update

Comment (spolti) [reply Dockerfile]: great, thanks @davidesalerno

**Pull request merged**